### PR TITLE
Add class doc to TPESampler.

### PR DESCRIPTION
**Pull request by ytsmiling**
## Motivation
The current class doc of TPE sampler misses the ` Args` section. Since the arguments are complicated and the names are confusing, it's better to have it.

## Description of the changes
Add the `Args` section to TPE's class doc.

### CODE REVIEW

File: optuna/samplers/tpe/sampler.py [hunk 1]
```diff
@@ -85,6 +85,37 @@ def objective(trial):
...
+            Number of steps required to run TPE. When the total number of finished trials
+            are less than this value, this sampler switches to random sampler.
+        n_ei_candidate:
+            Number of candidate samples used to calculate the estimated improvement.
+        gamma:
+            A function that takes the number of finished trials and returns the number
+            of trials to form a density function for samples with low grains.
+            See the original paper for more details.
+        weights:
+            A function that takes the number of finished trials and returns a weight for the trial.
```
Comment (HideakiImamura) [inline optuna/samplers/tpe/sampler.py:113]:
```suggestion
            A function that takes the number of finished trials and returns a weight for those trials.
```

**Comment by ytsmiling** [comment]
Thank you for your comments. I checked and committed all of them.

File: optuna/samplers/tpe/sampler.py [hunk 2]
```diff
@@ -86,6 +86,38 @@ def objective(trial):
...
+        n_ei_candidate:
+            Number of candidate samples used to calculate the expected improvement.
+        gamma:
+            A function that takes the number of finished trials and returns the number
+            of trials to form a density function for samples with low grains.
+            See the original paper for more details.
+        weights:
+            A function that takes the number of finished trials and returns a weight for them.
+            See `Making a Science of Model Search: Hyperparameter Optimization in Hundreds of
+            Dimensions for Vision Architectures <http://proceedings.mlr.press/v28/bergstra13.pdf>`
```
Comment (c-bata) [inline optuna/samplers/tpe/sampler.py:117]: Is this properly rendered as a link text? It looks invalid.

Comment (ytsmiling) [reply optuna/samplers/tpe/sampler.py]: Thank you for caching the error. I fixed the problem.

File: optuna/samplers/tpe/sampler.py [hunk 3]
```diff
@@ -86,6 +86,38 @@ def objective(trial):
             study = optuna.create_study(sampler=TPESampler())
             study.optimize(objective, n_trials=10)
 
+    Args:
+        consider_prior:
+            Enhance the stability of Parzen estimator by imposing a Gaussian prior when True. The
+            prior is only effective if the sampling distribution is either `UniformDistribution`,
```
Comment (sile) [inline optuna/samplers/tpe/sampler.py:92]: Please use the `:class:...` notation to create a link to the distribution doc.
```suggestion
            prior is only effective if the sampling distribution is either :class:`~optuna.distributions.UniformDistribution`,
```

File: optuna/samplers/tpe/sampler.py [hunk 4]
```diff
@@ -86,6 +86,38 @@ def objective(trial):
...
+            `DiscreteUniformDistribution`, `LogUniformDistribution`, or `IntUniformDistribution`.
+        prior_weight:
+            The weight of the prior. This argument is used in `UniformDistribution`,
+            `DiscreteUniformDistribution`, `LogUniformDistribution`, `IntUniformDistribution` and
```
Comment (sile) [inline optuna/samplers/tpe/sampler.py:96]: Ditto.

File: optuna/samplers/tpe/sampler.py [hunk 5]
```diff
@@ -86,6 +86,38 @@ def objective(trial):
             study = optuna.create_study(sampler=TPESampler())
             study.optimize(objective, n_trials=10)
 
+    Args:
+        consider_prior:
+            Enhance the stability of Parzen estimator by imposing a Gaussian prior when True. The
```
Comment (sile) [inline optuna/samplers/tpe/sampler.py:91]: Suggestion to align to the Optuna doc convention:
```suggestion
            Enhance the stability of Parzen estimator by imposing a Gaussian prior when :obj:`True`. The
```

**Pull request merged**